File: packages/etwin-api-types/src/lib/auth/auth-type.ts
import { TsEnumType } from "kryo/lib/ts-enum.js";

export enum AuthType {
  /**
   * Unauthenticated
   */
  Guest,

  /**
   * Authenticated as a user.
   */
  User,

  /**
   * System action
   */
  System,
}

export const $AuthType: TsEnumType<AuthType> = new TsEnumType<AuthType>({
  enum: AuthType,
});

File: clients/ruby/lib/etwin/auth/auth_type.rb
# typed: strict
# frozen_string_literal: true

module Etwin
  module Auth
    # Authentication type enum
    class AuthType < T::Enum
      extend T::Sig

      enums do
        Guest = new('Guest')
        User = new('User')
      end

      sig { returns(String) }
      def to_s
        T.cast(serialize, String)
      end

      sig { returns(String) }
      def inspect
        "AuthType(#{self})"
      end
    end
  end
end
